src/core/lib/surface/channel.cc

```diff
@@ -143,10 +144,15 @@ static grpc_core::UniquePtr<char> get_default_authority(
     } else if (0 == strcmp(input_args->args[i].key,
                            GRPC_SSL_TARGET_NAME_OVERRIDE_ARG)) {
       ssl_override = grpc_channel_arg_get_string(&input_args->args[i]);
+    } else if (0 == strcmp(input_args->args[i].key,
+                           GRPC_ALTS_TARGET_NAME_OVERRIDE_ARG)) {
+      alts_override = grpc_channel_arg_get_string(&input_args->args[i]);
     }
   }
   if (!has_default_authority && ssl_override != nullptr) {
     default_authority.reset(gpr_strdup(ssl_override));
+  } else if (!has_default_authority && alts_override != nullptr) {
```

Comment:
If the checks above stay here, this should just be combined with the check on line 152.

```
if (!has_default_authority && ssl_override != nullptr && alts_override != nullptr) {
```

Reply:
I've removed this code from `src/core/lib/surface/channel.cc`